specification/cdn/resource-manager/Microsoft.Cdn/2017-10-12/cdn.json

```diff
@@ -0,0 +1,3186 @@
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    },
+    "/providers/Microsoft.Cdn/edgenodes": {
+      "get": {
+        "tags": [
+          "Edgenodes"
+        ],
+        "description": "Edgenodes are the global Point of Presence (POP) locations used to deliver CDN content to end users.",
+        "operationId": "EdgeNodes_List",
+        "x-ms-examples": {
+          "EdgeNodes_List": {
+            "$ref": "./examples/EdgeNodes_List.json"
+          }
+        },
+        "parameters": [
+          {
+            "$ref": "#/parameters/apiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "OK. The request has succeeded.",
+            "schema": {
+              "$ref": "#/definitions/EdgenodeResult"
+            }
+          },
+          "default": {
+            "description": "CDN error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-pageable": {
+          "nextLinkName": "nextLink"
+        }
+      }
+    }
+  },
+  "definitions": {
+    "Profile": {
+      "description": "CDN profile is a logical grouping of endpoints that share the same settings, such as CDN provider and pricing tier.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/TrackedResource"
+        }
+      ],
+      "properties": {
+        "sku": {
+          "description": "The pricing tier (defines a CDN provider, feature list and rate) of the CDN profile.",
+          "$ref": "#/definitions/Sku"
+        },
+        "properties": {
+          "x-ms-client-flatten": true,
+          "$ref": "#/definitions/ProfileProperties"
+        }
+      },
+      "required": [
+        "sku"
+      ]
+    },
+    "ProfileProperties": {
+      "description": "The JSON object that contains the properties required to create a profile.",
+      "properties": {
+        "resourceState": {
+          "description": "Resource status of the profile.",
+          "readOnly": true,
+          "enum": [
+            "Creating",
+            "Active",
+            "Deleting",
+            "Disabled"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "ProfileResourceState",
+            "modelAsString": true
+          }
+        },
+        "provisioningState": {
+          "readOnly": true,
+          "description": "Provisioning status of the profile.",
+          "type": "string"
+        }
+      }
+    },
+    "ProfileListResult": {
+      "description": "Result of the request to list profiles. It contains a list of profile objects and a URL link to get the the next set of results.",
+      "properties": {
+        "value": {
+          "type": "array",
+          "readOnly": true,
+          "items": {
+      
```

Comment:
Consider making enums expandable if more could be added in the future. You can use "modeAsString".